autoextract_poet/items.py

```diff
@@ -7,12 +7,45 @@
 class Item:
 
     @classmethod
-    def from_dict(cls, item: Optional[Dict]):
-        return cls(**item) if item else None  # type: ignore
+    def from_dict(cls, item: dict):
+        return cls(**item)  # type: ignore
 
     @classmethod
-    def from_list(cls, items: List[Dict]):
-        return [cls.from_dict(item) for item in items if item]
+    def _get_attr_cls(cls, annotation):
+        """Inspect type annotation searching for a subclass of Item"""
+        if type(annotation) is type:
+            # Annotation is a class/type
+            return annotation if issubclass(annotation, Item) else None
+
+        # Annotation is probably a typing instance
+        for arg in annotation.__args__:
+            attr_cls = cls._get_attr_cls(arg)
+            if attr_cls:
+                return attr_cls
+
+    def __attrs_post_init__(self):
```

Comment:
I would move this code directly to the `from_dict` method. I think is good to have the default attrs initializer untouched, and use the `from_dict` to deserialize from dicts. Both interfaces are useful for the developer.

Reply:
Could you specify use-case differences between `from_dict` and `__init__` methods? I can only think of bypassing validation.